content-src/components/DiscoveryStreamComponents/TopSites/TopSites.jsx

```diff
@@ -31,8 +31,16 @@ export class _TopSites extends React.PureComponent {
       url: topSiteSpoc.url,
     };
 
-    const position = Math.min(topSites.length, spocPosition);
-    topSites[position] = link;
+    if (spocPosition === 0) {
```

Comment:
seeing a bug while pinning a non-spoc topsite when there is a spoc at position 0, looks like we need to bump up insert index to account for a spoc at zero position

Reply:
Ahhh. Good catch, but just to verify do we want this extra behavior? Because then there's an opportunity for SPOCs to appear in the middle of the row. @wolasi

Bug
![Bug](https://user-images.githubusercontent.com/25225547/63281436-93468780-c261-11e9-94a3-d7d1f7ca2681.gif)

Fix
![Bug Fix](https://user-images.githubusercontent.com/25225547/63281440-95104b00-c261-11e9-9922-83873868d18d.gif)

Reply:
Just realized too that this will happen in the end position also.... It would be easier if we have Pinned things as the highest priority then, and so the SPOC would go to the first available position that's not pinned.

![Screen Recording 2019-08-19 at 9 22 16 AM](https://user-images.githubusercontent.com/25225547/63282176-11eff480-c263-11e9-96e5-583949a2191c.gif)

Reply:
Good catch, let's fix both bugs if possible.